### PR TITLE
Fix/Logger memory leak

### PR DESCRIPTION
**Pull request by SNiukalov**
Fixes [#5239](https://adc.luxoft.com/jira/browse/FORDTCN-5239)

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
ATF scripts

### Summary
Updated log4cxx library to avoid memory leak and use weak pointer.
SDL rework to use log4cxx weak pointer.

##### Bug Fixes
* log4cxx memory accumulation 

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/include/application_manager/commands/command_impl.h [hunk 1]
```diff
@@ -201,7 +201,7 @@ class CommandImpl : public Command {
   CommandParametersPermissions removed_parameters_permissions_;
 
 #ifdef ENABLE_LOG
-  static log4cxx::LoggerPtr logger_;
+  static const log4cxx::LoggerWeakPtr logger_;
```
Comment (AByzhynar) [inline src/components/application_manager/include/application_manager/commands/command_impl.h:204]: @SNiukalov Can you please explain why did you make it static const? I mean what is the sense here: to create a static pointer to unchangeable value?

File: src/components/utils/src/push_log.cc [hunk 2]
```diff
@@ -100,7 +100,9 @@ void create_log_message_loop_thread() {
   }
 }
 
-void delete_log_message_loop_thread(log4cxx::LoggerPtr& logger) {
+void delete_log_message_loop_thread(const log4cxx::LoggerWeakPtr& del_logger) {
```
Comment (AByzhynar) [inline src/components/utils/src/push_log.cc:103]: @SNiukalov Bad parameter name. What does it mean? Delegating_logger? Delegated_logger? Deleting_logger? Deleted_logger? Deleicious_logger? Etc.

File: src/3rd_party/apache-log4cxx-0.10.0/src/main/cpp/fallbackerrorhandler.cpp [hunk 3]
```diff
@@ -53,8 +53,12 @@ void FallbackErrorHandler::error(const LogString& message,
                   +  message, e);
     LogLog::debug(LOG4CXX_STR("FB: INITIATING FALLBACK PROCEDURE."));
 
+    auto primary = primary_.lock();
+    auto backup = backup_.lock();
+
     for(size_t i = 0; i < loggers.size(); i++) {
-        LoggerPtr& l = (LoggerPtr&)loggers.at(i);
+        auto& logger = (LoggerWeakPtr&)loggers.at(i);
```
Comment (AByzhynar) [inline src/3rd_party/apache-log4cxx-0.10.0/src/main/cpp/fallbackerrorhandler.cpp:60]: @SNiukalov static_cast?

File: src/3rd_party/apache-log4cxx-0.10.0/src/main/cpp/propertyconfigurator.cpp [hunk 4]
```diff
@@ -342,23 +344,25 @@ void PropertyConfigurator::parseLogger(
                       + appenderName + LOG4CXX_STR("\"."));
         appender = parseAppender(props, appenderName);
 
-        if (appender != 0) {
-            logger->addAppender(appender);
+        if (false == appender.expired()) {
```
Comment (AByzhynar) [inline src/3rd_party/apache-log4cxx-0.10.0/src/main/cpp/propertyconfigurator.cpp:347]: @SNiukalov Use if (!appender.expired())

File: src/3rd_party/apache-log4cxx-0.10.0/src/main/include/log4cxx/varia/fallbackerrorhandler.h [hunk 5]
```diff
@@ -38,9 +38,9 @@ namespace log4cxx {
             public virtual spi::ErrorHandler,
             public virtual helpers::Object {
             private:
-                AppenderPtr backup;
-                AppenderPtr primary;
-                std::vector<LoggerPtr> loggers;
+                AppenderWeakPtr backup_;
+                AppenderWeakPtr primary_;
+                std::vector<LoggerWeakPtr> loggers;
```
Comment (AByzhynar) [inline src/3rd_party/apache-log4cxx-0.10.0/src/main/include/log4cxx/varia/fallbackerrorhandler.h:43]: @SNiukalov Add postfix underscore to `loggers` variable

File: src/3rd_party/apache-log4cxx-0.10.0/src/main/cpp/telnetappender.cpp [hunk 6]
```diff
@@ -159,7 +159,7 @@ void TelnetAppender::append(const spi::LoggingEventPtr& event, Pool& p) {
 
     if (count > 0) {
         LogString msg;
-        this->layout->format(msg, event, pool);
+        this->layout->format(msg, event, p);
```
Comment (AByzhynar) [inline src/3rd_party/apache-log4cxx-0.10.0/src/main/cpp/telnetappender.cpp:162]: @SNiukalov I would prefer previous naming. `pool` is better than just `p`

**Comment by SNiukalov** [comment]
Moving to LuxoftSDL

**Pull request closed without merging**